livekit-agents/livekit/agents/beta/workflows/dtmf_inputs.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import annotations
+
+import logging
+
+from typing import Callable, Optional
+
+from livekit import rtc
+from livekit.agents.voice.events import AgentStateChangedEvent, UserStateChangedEvent
+
+from ... import function_tool
+from ...job import get_job_context
+from ...llm.chat_context import ChatContext
+from ...types import NOT_GIVEN, NotGivenOr
+from ...utils.aio.debounce import Debounced, debounced
+from ...voice.agent import AgentTask
+from ..tools.dtmf import DtmfEvent, format_dtmf
+
+logger = logging.getLogger("dtmf-inputs")
+
+
+class GetDtmfTask(AgentTask[Optional[str]]):
```

Comment:
Can we create a Result dataclass like:
https://github.com/livekit/agents/blob/a4ad0dba2137e1b972dc97b38c41502817aa97cb/livekit-agents/livekit/agents/beta/workflows/email_address.py#L23

Also in case we have an error, let's raise ToolError (`AgentTask.complete` also accepts Error as the complete value) instead of None.